tests/tests_suite_test.go

```diff
@@ -0,0 +1,165 @@
+package tests
+
+import (
+	"context"
+	migapi "github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"log"
+	"os"
+	"path/filepath"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"testing"
+	"time"
+
+	// +kubebuilder:scaffold:imports
+)
+
+func TestMigmigration(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "E2E Suite")
+}
+
+var sourceClient *kubernetes.Clientset
+var hostClient client.Client
+var hostCfg *rest.Config
+var sourceCfg *rest.Config
+var dynamicClient dynamic.Interface
+var controllerCR schema.GroupVersionResource
+var err error
+var shouldCreate bool
+
+var _ = BeforeSuite(func() {
+	hostCfg, err = clientcmd.BuildConfigFromFlags("", filepath.Join(os.Getenv(HOSTCONFIG)))
+	if err != nil {
+		log.Println(err)
+	}
+
+	err = migapi.AddToScheme(scheme.Scheme)
+	hostClient, err = client.New(hostCfg, client.Options{Scheme: scheme.Scheme})
+	if err != nil {
+		log.Println(err)
+	}
+	sourceCfg, err = clientcmd.BuildConfigFromFlags("", filepath.Join(os.Getenv(SOURCECONFIG)))
+	if err != nil {
+		log.Println(err)
+	}
+	sourceClient, err = kubernetes.NewForConfig(sourceCfg)
+	if err != nil {
+		log.Println(err)
+	}
+
+	dynamicClient, err = dynamic.NewForConfig(hostCfg)
+	if err != nil {
+		log.Println(err)
+	}
+	controllerCR = schema.GroupVersionResource{
+		Group:    "migration.openshift.io",
+		Version:  "v1alpha1",
+		Resource: "migrationcontrollers",
+	}
+
+	controller, err := dynamicClient.Resource(controllerCR).Namespace(MigrationNamespace).Get(context.TODO(), MigrationController, metav1.GetOptions{})
+	if err != nil {
+		log.Println(err)
```

Comment:
`errors.IsNotFound(err)`

using the above error check will be a more definitive case of when to create the controller and when to fail because the error is more fatal IMO

Reply:
+1